includes/options/class-amp-options-menu.php

```diff
@@ -1,54 +1,184 @@
 
+	/**
+	 * Initialize.
+	 */
 	public function init() {
 		add_action( 'admin_post_amp_analytics_options', 'AMP_Options_Manager::handle_analytics_submit' );
-
 		add_action( 'admin_menu', array( $this, 'add_menu_items' ) );
 	}
 
+	/**
+	 * Add menu.
+	 */
 	public function add_menu_items() {
+
 		add_menu_page(
 			__( 'AMP Options', 'amp' ),
 			__( 'AMP', 'amp' ),
 			'manage_options',
-			$this->menu_slug,
-			array( $this->menu_page, 'render' ),
+			AMP_Options_Manager::OPTION_NAME,
+			array( $this, 'render_screen' ),
 			self::ICON_BASE64_SVG
 		);
 
+		add_submenu_page(
+			AMP_Options_Manager::OPTION_NAME,
+			__( 'AMP Settings', 'amp' ),
+			__( 'General', 'amp' ),
+			'manage_options',
+			AMP_Options_Manager::OPTION_NAME
+		);
+
+		add_settings_section(
+			'post_types',
+			false,
+			'__return_false',
+			AMP_Options_Manager::OPTION_NAME
+		);
+		add_settings_field(
+			'supported_post_types',
+			__( 'Post Type Support', 'amp' ),
+			array( $this, 'render_post_types_support' ),
+			AMP_Options_Manager::OPTION_NAME,
+			'post_types'
+		);
+
 		$submenus = array(
-			new AMP_Analytics_Options_Submenu( $this->menu_slug ),
+			new AMP_Analytics_Options_Submenu( AMP_Options_Manager::OPTION_NAME ),
 		);
 
 		// Create submenu items and calls on the Submenu Page object to render the actual contents of the page.
 		foreach ( $submenus as $submenu ) {
-			$submenu->init( $this->menu_slug );
+			$submenu->init();
 		}
-
-		$this->remove_toplevel_menu_item();
 	}
 
-	// Helper function to avoid having the top-level menu as
-	// the first menu item
-	function remove_toplevel_menu_item() {
-		global $submenu;
-		if ( isset( $submenu['amp-plugin-options'][0] ) ) {
-			unset( $submenu['amp-plugin-options'][0] );
+	/**
+	 * Check for errors with updating the supported post types.
+	 *
+	 * @since 0.6
+	 */
+	protected function check_supported_post_type_update_errors() {
```

Comment:
Yes, it's still needed because the error checking can only be done for these errors after WordPress finishes initializing, since we have to wait for themes and plugins to have called `add_post_type_support()` or `remove_post_type_support()` in order to detect whether there has been an error scenario.

Reply:
So my bad, what I meant was why having this method here (structurally) as opposed to moving it to the `AMP_Options_Manager` class where the other error handlings are.